DragonSlayer.cpp

```diff
@@ -1,27 +1,49 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Character.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer (std::string name_, int hp, int armor) : Character ( hp, armor, 4 ), name( name_ )
+{
+    std::cout << getName() << std::endl;
+    helpfulItems = makeHelpfulItems( rand() % 4 );
+    defensiveItems = makeDefensiveItems( rand() % 3 );
+}
+
+DragonSlayer::~DragonSlayer(){}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        //assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //
+        dragonSword->use(this);
+        dragonSword.reset();
+
         while( dragon->getHP() > 0 )
         {
           dragon->takeDamage(attackDamage);
         }
+        //reset attack damage after killing dragon
+        attackDamage = this->getInitialAttackDamage();
```

Comment:
`initialAttackDamage` is actually protected.  you don't need a member function to access it in derived classes.
- [ ] remove getter usage.  assign directly.

Reply:
whenever I try to access `initialAttackDamage ` directly I get a "Can't access private member variable" error